pandas/_libs/internals.pyx

```diff
@@ -996,3 +1000,59 @@ cdef class BlockValuesRefs:
                 return self._has_reference_maybe_locked()
         ELSE:
             return self._has_reference_maybe_locked()
+
+
+cdef extern from "Python.h":
+    """
+    #if PY_VERSION_HEX < 0x030E0000
+    int __Pyx_PyUnstable_Object_IsUniqueReferencedTemporary(PyObject *ref)
+    {
+        return 0;
+    }
+    #else
+    #define __Pyx_PyUnstable_Object_IsUniqueReferencedTemporary \
+        PyUnstable_Object_IsUniqueReferencedTemporary
+    #endif
+    """
+    int PyUnstable_Object_IsUniqueReferencedTemporary\
+        "__Pyx_PyUnstable_Object_IsUniqueReferencedTemporary"(object o) except -1
+
+
+cdef inline bint _is_unique_referenced_temporary(object obj) except -1:
+    if PY_VERSION_HEX >= 0x030E0000:
+        return PyUnstable_Object_IsUniqueReferencedTemporary(obj)
+    else:
+        return sys.getrefcount(obj) <= 1
+
+
+# # Python version compatibility for PyUnstable_Object_IsUniqueReferencedTemporary
+# IF PY_VERSION_HEX >= 0x030E0000:
+#     # Python 3.14+ has PyUnstable_Object_IsUniqueReferencedTemporary
+#     cdef inline bint _is_unique_referenced_temporary(object obj) except -1:
+#         return PyUnstable_Object_IsUniqueReferencedTemporary(obj)
+# ELSE:
+#     # Fallback for older Python versions using sys.getrefcount
+#     cdef inline bint _is_unique_referenced_temporary(object obj) except -1:
+#         # sys.getrefcount includes the reference from getrefcount itself
+#         # So if refcount is 2, it means only one external reference exists
+#         return sys.getrefcount(obj) == 2
+
+
+# @cython.auto_pickle(False)
+cdef class SetitemMixin:
+
+    def __setitem__(self, key, value):
```

Comment:
can this be made cdef?  small perf improvement i think.  downside is i think it becomes harder for subclassers who want to override it.  But after this PR, they should be overriding _setitem instead

Reply:
No, special (dunder) methods need to be specified as `def` (That's also what I would have expected since this method should be callable from python, but even cpdef gives a cython compilation error "Special methods must be declared with 'def', not 'cdef'")